subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestLauncherTestSpecCrossVersionSpec.groovy

```diff
@@ -37,7 +37,7 @@ class TestLauncherTestSpecCrossVersionSpec extends TestLauncherSpec {
     def "older Gradle versions ignore withTestsFor calls"() {
         when:
         launchTests { TestLauncher launcher ->
-            launcher.withTestsFor { TestSpec spec ->
+            launcher.withTestsFor { TestSpecFactory spec ->
                 spec.forTaskPath(':secondTest').includeClass('example.MyTest')
             }
```

Comment:
I'm not really happy with the new (and renamed) `TestSpecFactory` API. Maybe it should be more explicit with object creation and the state change: 

```groovy
TestLauncher launcher = ...
launcher withTestsFor { TestSpecRegisty specs -> 
    specs.add(specs.builder(':taskPath').includeClass('example.MyTest').build())
}
```

wdyt?